### PR TITLE
fix: [#2745] Fix current eslint warnings - botbuilder-dialogs-adaptive library - src & conditions

### PR DESCRIPTION
**Pull request by NahuelPerales**
Adresses # 2745
#minor

## Description
When reviewing the botbuilder-js project it was found that eslint is showing several warnings for the botbuilder-dialogs-adaptive library.

## Specific Changes

- Replaced strings quotes with the correct one depending of the case
- Replaced hasOwnProperty with Object.prototype.hasOwnProperty.call
- Adjusted formatting
- Limited importing scopes
- Added missing jsdocs documentation for parameters and returns for functions
- Removed unused variables

## Testing
Eslint with no warnings and tests passed
image

### CODE REVIEW

File: libraries/botbuilder-dialogs-adaptive/etc/botbuilder-dialogs-adaptive.api.md [hunk 1]
```diff
@@ -180,6 +180,7 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> im
     recognizer?: Recognizer;
     repromptDialog(context: DialogContext | TurnContext, instance: DialogInstance): Promise<void>;
     resumeDialog(dc: DialogContext, _reason?: DialogReason, _result?: any): Promise<DialogTurnResult>;
+    // Warning: (ae-setter-with-docs) The doc comment for the property "schema" must appear on the getter, not the setter.
```
Comment (JuanAr) [inline libraries/botbuilder-dialogs-adaptive/etc/botbuilder-dialogs-adaptive.api.md:183]: I don't know why this file is changed. Is this part of the ESLint work?

File: libraries/botbuilder-dialogs-adaptive/package.json [hunk 2]
```diff
@@ -28,6 +28,12 @@
     }
   },
   "dependencies": {
+    "@microsoft/recognizers-text-sequence": "~1.1.4",
```
Comment (JuanAr) [inline libraries/botbuilder-dialogs-adaptive/package.json:31]: I don't know why this file is changed. Is this part of the ESLint work?

**Review by JuanAr — CHANGES_REQUESTED** [review]
I notice some changes not really related to ESLint. Can you double check.

**Review by JuanAr — APPROVED** [review]
LGTM

File: libraries/botbuilder-dialogs-adaptive/package.json (hunk 2 above)
Comment (sw-joelmut) [reply libraries/botbuilder-dialogs-adaptive/package.json:31]: What is the response for this?, we should add it to the PR's description

File: libraries/botbuilder-dialogs-adaptive/src/actionContext.ts [hunk 3]
```diff
@@ -62,6 +66,8 @@ export class ActionContext extends DialogContext {
 
     /**
      * Applies any queued up changes.
+     *
+     * @returns True if succeed otherwise false
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/actionContext.ts:70]:
```suggestion
     * @returns True if succeed, otherwise false.
```

File: libraries/botbuilder-dialogs-adaptive/src/adaptiveBotComponent.ts [hunk 4]
```diff
@@ -121,7 +121,14 @@ import {
     UrlEntityRecognizer,
 } from './recognizers';
 
+/**
+ *
+ */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/adaptiveBotComponent.ts:126]: Should have a description

File: libraries/botbuilder-dialogs-adaptive/src/adaptiveBotComponent.ts [hunk 5]
```diff
@@ -121,7 +121,14 @@ import {
...
 export class AdaptiveBotComponent extends BotComponent {
+    /**
+     * @param services A list of services.
+     * @param _configuration The configuration.
+     */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/adaptiveBotComponent.ts:131]: Use description and params from DotNet, i also could be defined in the BotComponent class and use the `@inheritdoc`

File: libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts [hunk 6]
```diff
@@ -147,6 +147,9 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> im
      */
     public defaultResultProperty = 'dialog.result';
 
+    /**
+     *
+     */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts:152]: same

File: libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts [hunk 7]
```diff
@@ -160,6 +163,10 @@ export class AdaptiveDialog<O extends object = {}> extends DialogContainer<O> im
         return this.dialogSchema ? this.dialogSchema.schema : undefined;
     }
 
+    /**
+     * @param property The key of the conditional selector configuration.
+     * @returns The converter for the selector configuration.
+     */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/adaptiveDialog.ts:169]: Same case mentioned in other PR

File: libraries/botbuilder-dialogs-adaptive/src/conditions/onCondition.ts [hunk 8]
```diff
@@ -91,6 +92,7 @@ export class OnCondition extends Configurable implements DialogDependencies, OnC
 
     /**
      * @protected
+     * @returns The action scope.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/conditions/onCondition.ts:95]: Use the description and return from dotnet

File: libraries/botbuilder-dialogs-adaptive/src/conditions/onCondition.ts [hunk 9]
```diff
@@ -180,6 +190,8 @@ export class OnCondition extends Configurable implements DialogDependencies, OnC
 
     /**
      * Get child dialog dependencies so they can be added to the containers dialogset.
+     *
+     * @returns The dependencies.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/conditions/onCondition.ts:194]: Use the return from dotnet

File: libraries/botbuilder-dialogs-adaptive/src/entityAssignment.ts [hunk 10]
```diff
@@ -94,6 +94,9 @@ export class EntityAssignment implements EntityAssignmentConfiguration {
      */
     public expectedProperties: string[];
 
+    /**
+     * @param assignment The assigment
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/entityAssignment.ts:98]: The entity assignment properties to set for this instance.

File: libraries/botbuilder-dialogs-adaptive/src/entityAssignment.ts [hunk 11]
```diff
@@ -145,6 +150,8 @@ export class EntityAssignment implements EntityAssignmentConfiguration {
 
     /**
      * Print an assignment as a string.
+     *
+     * @returns The assigment as string.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/entityAssignment.ts:154]: Use the one from dotnet

File: libraries/botbuilder-dialogs-adaptive/src/entityAssignmentComparer.ts [hunk 12]
```diff
@@ -8,15 +8,26 @@
 import { AdaptiveEvents } from './adaptiveEvents';
 import { EntityAssignment } from './entityAssignment';
 
+/**
+ *
+ */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/entityAssignmentComparer.ts:13]: same

File: libraries/botbuilder-dialogs-adaptive/src/entityAssignmentComparer.ts [hunk 13]
```diff
@@ -8,15 +8,26 @@
...
 export class EntityAssignmentComparer {
     private static eventPreference = [
         AdaptiveEvents.assignEntity,
         AdaptiveEvents.chooseProperty,
         AdaptiveEvents.chooseEntity,
     ];
 
+    /**
+     * @param operationPreference The operation preference.
+     */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/entityAssignmentComparer.ts:23]: Use description and param from dotnet

File: libraries/botbuilder-dialogs-adaptive/src/entityAssignmentComparer.ts [hunk 14]
```diff
@@ -8,15 +8,26 @@
...
     constructor(private operationPreference: string[]) {}
 
+    /**
+     * @param x First entity assigment to compare.
+     * @param y Second entity assigment to compare.
+     * @returns Result of the comparison.
+     */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/entityAssignmentComparer.ts:30]: Use jsdoc from dotnet

File: libraries/botbuilder-dialogs-adaptive/src/entityAssignments.ts [hunk 15]
```diff
@@ -17,15 +17,21 @@ export interface EntityAssignmentsConfiguration {
...
+    /**
+     * @param assignments Entities to assign.
+     */
     constructor(public assignments: EntityAssignment[] = []) {}
 
     /**
      * Read entity event queue from memory.
+     *
      * @param actionContext Memory context.
+     * @returns The entity assigments.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/entityAssignments.ts:34]: use from dotnet

File: libraries/botbuilder-dialogs-adaptive/src/entityAssignments.ts [hunk 16]
```diff
@@ -56,7 +63,9 @@ export class EntityAssignments implements EntityAssignmentsConfiguration {
 
     /**
      * Remove the current event and update the memory.
+     *
      * @param actionContext Memory context.
+     * @returns The assigment.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/entityAssignments.ts:68]: use from dotnet

File: libraries/botbuilder-dialogs-adaptive/src/entityInfo.ts [hunk 17]
```diff
@@ -89,34 +89,42 @@ export interface NormalizedEntityInfos {
 export class EntityInfo {
     /**
      * Print an entity as a string.
+     *
      * @param source Source entity.
+     * @returns A string representing the entity.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/entityInfo.ts:94]: Use from dotnet

File: libraries/botbuilder-dialogs-adaptive/src/languageGenerationBotComponent.ts [hunk 18]
```diff
@@ -6,7 +6,14 @@ import { BotComponent } from 'botbuilder';
 import { ComponentDeclarativeTypes } from 'botbuilder-dialogs-declarative';
 import { Configuration, ServiceCollection } from 'botbuilder-dialogs-adaptive-runtime-core';
 
+/**
+ *
+ */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/languageGenerationBotComponent.ts:11]: same

File: libraries/botbuilder-dialogs-adaptive/src/languageGenerationBotComponent.ts [hunk 19]
```diff
@@ -6,7 +6,14 @@ import { BotComponent } from 'botbuilder';
...
 export class LanguageGenerationBotComponent extends BotComponent {
+    /**
+     * @param services The list of services.
+     * @param _configuration The configuration.
+     */
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/languageGenerationBotComponent.ts:16]: This is the same case mentioned before

File: libraries/botbuilder-dialogs-adaptive/src/skillExtensions.ts [hunk 20]
```diff
@@ -25,8 +25,10 @@ export const skillConversationIdFactoryKey = Symbol('SkillConversationIdFactory'
 export class SkillExtensions {
     /**
      * Configures the skill client to use.
+     *
      * @param dialogManager The dialog manager to add skill client to.
      * @param skillClient The skill client to be added.
+     * @returns The dialog manager based on the skill client.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/skillExtensions.ts:31]: This return could be changed to something like 
`The existing dialog manager object with the new skill client set into the state.`

File: libraries/botbuilder-dialogs-adaptive/src/skillExtensions.ts [hunk 21]
```diff
@@ -35,8 +37,10 @@ export class SkillExtensions {
 
     /**
      * Configures the skill conversation id factory to use.
+     *
      * @param dialogManager The dialog manager to add skill conversation id factory to.
      * @param skillConversationIdFactory The skill conversation id factory to be added.
+     * @returns The dialog manager based on the skill factory.
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs-adaptive/src/skillExtensions.ts:43]: Similar as above

File: libraries/botbuilder-dialogs/src/dialogContext.ts [hunk 22]
```diff
@@ -2,653 +2,654 @@
...
-            }
-
-            // Pop dialog off stack
-            this.stack.pop();
-
-            this.state.setValue(TurnPath.lastResult, result);
-        }
-    }
-}
+ import { Activity, TurnContext, TurnContextStateCollection } from 'botbuilder-core';
```
Comment (sw-joelmut) [inline libraries/botbuilder-dialogs/src/dialogContext.ts:5]: This entire file is not properly fixed, why is this file changed from another library?

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

File: libraries/botbuilder-dialogs-adaptive/package.json (hunk 2 above)
Comment (sw-joelmut) [reply libraries/botbuilder-dialogs-adaptive/package.json:31]: ESLint suggests moving them from devDependencies to dependencies

**Review by sw-joelmut — APPROVED** [review]

**Comment by sw-joelmut** [comment]
Promoted `https://github.com/microsoft/botbuilder-js/pull/4069`
